### ISSUE TITLE
Reporting Challenge

### ISSUE THREAD

**Issue by Lesego-Serobatse** [Praneetbond/8Week_SQL_Challenge#1]
Hi Praneetbond, I like your work, you have done really well. 
I'd like to understand the reporting challenge on Balanced Tree Clothing challenge, I am not sure what is really required here. The part that says 'single sql script', what does that mean. And how is it possible for a report to have multiple table outputs, isnt a report supposed to be in a single table, may you kindly explain your understanding of what we are supposed to do in this task. Thank you.

**Comment by Praneetbond** [comment]
Hi Lesego-Serobatse,
Let me try a simple way, based on what I understand, the term 'single SQL script' typically refers to a collection of queries combined into a script. However, in this context, it appears to mean that a SQL query is designed to retrieve specific data, and even when that data is updated with additional information in the following month, the same SQL query can still provide the desired results without requiring any modifications. And regarding the aspect of 'Reporting' my approach involves creating separate tables for distinct KPIs, these tables can be used to generate essential queries. The question also implies that it's possible to generate numerous tables as needed and with minor adjustments, these tables can consistently produce the same KPIs every month, despite changes in the underlying data. Hope I answered your question else just ask

Thank you for your interest

**Comment by Lesego-Serobatse** [comment]
You hit the nail on the head Praneetbond, 'collection of queries combined into a (single) script'. Is it possible(impossible is nothing, I know) to do this in this task, if you were strickly to adhere to the requirements, of combining all this queries into a single query, how would you go about it (a thought of using CTE's just crossed my mind as I am responding to you). Let me try using a string of CTE's and output the results on different tables.  I see you have used table expressions, but you have multiple scripts. I'll try to merge everything, let me see if its possible.
Okay thank you Praneetbond, thank you very much

**Comment by Praneetbond** [comment]
I tried different ways to solve it but in the end, this approach looks right and easy to understand for everyone. I will try again if a single query is possible.

Happy to help a positive person like you. Hope you got something

**Comment by Lesego-Serobatse** [comment]
I tried and failed for now. So I changed lanes and adopted a lazy thinking approach, so I just dumped every query (solutions to questions) in a Procedure with a @month_number parameter. So I can run the:
**_EXEC PROCEDURE Month_Report @month_number = ...(whatever month)_**  every month, lol😂🤣.
So I have achieved the requirements for now, lol, even though I do not think it's proper thinking to the solution, plus there are multiple output tables, lol, talk about lazy thinking achieving the requirements. So the results will indeed serve as a report because the table outputs are simultaneous, and they look like a report(me, trying to convince myself that I did the right thing).

Anyway thanks for your input, I hope we'll find a proper solution to this difficult problem. 
Thanks Praneet
![SoExcited~GIF (2)](https://github.com/Praneetbond/8Week_SQL_Challenge/assets/90797415/0f1df585-d1fd-4c10-a993-a63d40f8c3c6)
